### PR TITLE
Implemented Google Recaptcha

### PR DESCRIPTION
**Pull request by rashmi-91**

### CODE REVIEW

File: src/api/routes/visitors.js [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
 import sendEmail from "../../mailing"
 import routeHandler from "../routeHandler"
+import axios from 'axios';
 
 express.post('/api/contactForm', routeHandler(async (request, response) => {
 
-    const { name = '', country = '', phone = '', email = '', message = '' } = request.body;
+    const { name = '', country = '', phone = '', email = '', message = '', 'g-recaptcha-response': grecaptcharesponse } = request.body;
+
+    const recaptchaSecretKey = "6LdyBuQZAAAAAPrDOy-hjdQXMW4jzlKon8SCay3F";
```
Comment (ItamarGronich) [inline src/api/routes/visitors.js:10]: Move to `secrets.js` that's the place for secret keyes.

File: src/api/routes/visitors.js [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+    const googleVerifyUrl = `https://www.google.com/recaptcha/api/siteverify`;
```
Comment (ItamarGronich) [inline src/api/routes/visitors.js:11]: Move to `cfg/default.js` that's the place for any configurations like this.

File: src/api/routes/visitors.js [hunk 3]
```diff
@@ -14,7 +43,7 @@ express.post('/api/contactForm', routeHandler(async (request, response) => {
 
     try {
         await sendEmail({
-            to: 'BuizDev@PsySession.com',
+            to: 'support@PsySession.com',
```
Comment (ItamarGronich) [inline src/api/routes/visitors.js:46]: Can use that value from the config `supportEmail`.

**Review by ItamarGronich — CHANGES_REQUESTED** [review]
Did you run the flow once end-to-end and tested that it works?

File: src/api/routes/visitors.js (hunk 1 above)
Comment (ItamarGronich) [reply src/api/routes/visitors.js:10]: Note that once you do, before you deploy you have to `ssh` to production and modify the secrets file there and only then it's ok to deploy.
If you don't your code will look for that secret but it won't be there.

Also, please be sure to update the secrets file on google docs so new developers that download the file will have the updated secrets file.
Please also notify in the slack channels that you modified the secrets file so everyone working on the project would know to re-download the secrets file.

to sum it up:
1. add to secrets file locally, import it here and check that it works.
2. before deploy - modify the secrets file in production.
3. after deploy - modify the secrets file in google docs and let everyone know that you modified it.

File: src/api/routes/visitors.js (hunk 3 above)
Comment (rashmi-91) [reply src/api/routes/visitors.js:46]: import express from "../express"
import sendEmail from "../../mailing"
import routeHandler from "../routeHandler"
import axios from 'axios';
import cfg from "../../cfg";

express.post('/api/contactForm', routeHandler(async (request, response) => {

    const { name = '', country = '', phone = '', email = '', message = '', 'g-recaptcha-response': grecaptcharesponse } = request.body;

    const recaptchaSecretKey = cfg.googleCaptcha.secret;
    const googleVerifyUrl = `https://www.google.com/recaptcha/api/siteverify`;
    
    if (!(email || phone)) {
        response.status(422);
        response.json({})
        return;
    }

    try {
        const result = await axios({
            method: 'post',
            url: googleVerifyUrl,
            params: {
                secret: recaptchaSecretKey,
                response: grecaptcharesponse
            }
        });

        const data = result.data;              

        if (!data.success) {
            response.json("Please verify captcha");
            return;
        }

    }
    catch (e) {
        response.status(500);
        response.json(e);
        return;
    }       
   
    try {
        await sendEmail({
            to: cfg.supportEmail.emailId,
            subject: 'User submitted contact form',
            html: `
                <table>
                      <thead>
                            <th>Contact Form Details</th>
                      </thead>
                      <tbody>
                            <tr>
                                <td>Name:</td><td>${name}</td>
                            </tr>
                            <tr>
                                <td>Country:</td><td>${country}</td>
                            </tr>
                            <tr>
                                <td>Phone:</td><td>${phone}</td>
                            </tr>
                            <tr>
                                <td>Email:</td><td><a href="mailto:${email}">${email}</a></td>
                            </tr>
                            <tr>
                                <td>Message:</td><td>${message}</td>
                            </tr>
                      </tbody>
                </table>
            `,
        });
        response.status(200);
    } catch (e) {
        response.status(500);
        response.json(e);
    }

    response.json({});

}, {
    enforceLogin: false
}));

**Review by ItamarGronich — APPROVED** [review]

**Comment by ItamarGronich** [comment]
please merge squash and also delete branch after you merge this so the information about the secret key would be deleted from git.
